flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/coordinator/SchemaRegistryRequestHandler.java

```diff
@@ -348,6 +348,12 @@ public void close() throws IOException {
         if (schemaChangeThreadPool != null) {
             schemaChangeThreadPool.shutdown();
         }
+
+        try {
+            metadataApplier.close();
+        } catch (Exception e) {
+            throw new IOException("Failed to close metadata applier.", e);
+        }
```

Comment:
I've tested locally with a dummy MetadataApplier that always throws exception but gets no output. Seems OperatorCoordinators will be closed by calling `OperatorCoordinatorHandler#disposeAllOperatorCoordinators`, where exceptions thrown here are ignored without being logged:

```java
// DefaultOperatorCoordinatorHandler.java
@Override
public void disposeAllOperatorCoordinators() {
    coordinatorMap.values().forEach(IOUtils::closeQuietly);
}

// IOUtils.java
public static void closeQuietly(AutoCloseable closeable) {
    try {
        if (closeable != null) {
            closeable.close();
        }
    } catch (Throwable ignored) {
    }
}
```

Maybe we can add some logging message if metadata appliers are not closed as expected?

Reply:
Good catch 👍 Thanks for the detailed explanation.